lectures/equalizing_difference.md

```diff
@@ -242,75 +244,30 @@ For a given instance of the class, we want to recompute $\phi$ when one paramete
 Let's do an example.
 
 ```{code-cell} ipython3
-# ϕ_R
-def ϕ_R(mc, R_new):
-    mc_new = equalizing_diff(R_new, mc.T, mc.γ_h, mc.γ_c, mc.w_h0, mc.D, mc.π)
-    return mc_new.compute_gap()
-
-ϕ_R = np.vectorize(ϕ_R)
-
-# ϕ_γh
-def ϕ_γh(mc, γh_new):
-    mc_new = equalizing_diff(mc.R, mc.T, γh_new, mc.γ_c, mc.w_h0, mc.D, mc.π)
-    return mc_new.compute_gap()
-
-ϕ_γh = np.vectorize(ϕ_γh)
-
-# ϕ_γc
-def ϕ_γc(mc, γc_new):
-    mc_new = equalizing_diff(mc.R, mc.T, mc.γ_h, γc_new, mc.w_h0, mc.D, mc.π)
-    return mc_new.compute_gap()
+ex1 = create_edm()
+gap1 = compute_gap(ex1)
 
-ϕ_γc = np.vectorize(ϕ_γc)
-
-# ϕ_π
-def ϕ_π(mc, π_new):
-    mc_new = equalizing_diff(mc.R, mc.T, mc.γ_h, mc.γ_c, mc.w_h0, mc.D, π_new)
-    return mc_new.compute_gap()
-
-ϕ_π = np.vectorize(ϕ_π)
-```
-
-```{code-cell} ipython3
-# set benchmark parameters
-R = 1.05
-T = 40
-γ_h, γ_c = 1.01, 1.01
-w_h0 = 1
-D = 10
-
-# create an instance
-ex1 = equalizing_diff(R=R, T=T, γ_h=γ_h, γ_c=γ_c, w_h0=w_h0, D=D)
-gap1 = ex1.compute_gap()
-
-print(gap1)
+gap1
 ```
 
 Let's not charge for college and recompute $\phi$.
 
 The initial college wage premium should go down.
 
-
- 
-
 ```{code-cell} ipython3
 # free college
-ex2 = equalizing_diff(R, T, γ_h, γ_c, w_h0, D=0)
-gap2 = ex2.compute_gap()
-print(gap2)
+ex2 = create_edm(D=0)
+gap2 = compute_gap(ex2)
+gap2
 ```
 
-
-
 Let us construct some graphs that show us how the initial college-high-school wage ratio $\phi$ would change if one of its determinants were to change. 
 
-Let's start with the gross interest rate $R$.  
-
-
+Let's start with the gross interest rate $R$.
 
 ```{code-cell} ipython3
 R_arr = np.linspace(1, 1.2, 50)
-plt.plot(R_arr, φ_R(ex1, R_arr))
+plt.plot(R_arr, compute_gap(create_edm(R=R_arr)))
```

Comment:
Wow, this actually works??  That's amazing.  So   `create_edm(R=R_arr)` returns a numpy array of namedtuples??

This definitely needs some comments.

Reply:
I think this is passing the model with parameter `R` as a `numpy` array into `compute_gap` but not creating a `numpy` array of `namedtuple`. 

The `R_arr` is then unpacked in the `compute_gap` function, and the operation on the array is broadcasted within the function.

Reply:
Please see my response below.